openssh/auth.sls

```diff
@@ -38,12 +38,18 @@ include:
 {{ print_name(identifier, key) }}:
   ssh_auth.present:
     {{ print_ssh_auth(identifier, key) }}
+    {%- if 'sshd_config' in pillar and 'AuthorizedKeysFile' in pillar['sshd_config'] %}
+    - config: {{ pillar['sshd_config']['AuthorizedKeysFile'] }}
+    {% endif -%}
     - require:
       - service: {{ openssh.service }}
     {%- else %}
 {{ print_name(identifier, key) }}:
   ssh_auth.absent:
     {{ print_ssh_auth(identifier, key) }}
+    {%- if 'sshd_config' in pillar and 'AuthorizedKeysFile' in pillar['sshd_config'] %}
+    - config: {{ pillar['sshd_config']['AuthorizedKeysFile'] }}
```

Comment:
You need to put the `{{ ... }}` part between single quotes, like this: `'{{ ... }}'`

The reason is a bit complex:

- The `%` character has a special meaning in YAML
- On line 26 of `pillar.example`, the `%` sign is quoted. So it becomes a normal character.
- The whole line, without quotes, gets saved as the value of `pillar['sshd_config']['AuthorizedKeysFile']`
- However in the `auth.sls` file, the content of the above gets inlined _without quotes_, resulting in (after Jinja rendering):

        - config: %h/.ssh/authorized_keys

- And this last construct -- which has an unquoted `%` sign -- makes YAML go haywire.